### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by bernhardmgruber**
* made struct with a constant just a constant
* refactored iostream usage
* merged all *_impl headers into their declaring headers
* including mallocMC.hpp instead of individual headers in tests and examples

### CODE REVIEW

File: src/include/mallocMC/distributionPolicies/XMallocSIMD.hpp [hunk 1]
```diff
@@ -64,8 +73,127 @@ namespace mallocMC
          *        default configuration. The default can be obtained through
          *        XMallocSIMD<>::Properties
          */
-        template<typename T_Config = XMallocSIMDConf::DefaultXMallocConfig>
```
Comment (psychocoderHPC) [inline src/include/mallocMC/distributionPolicies/XMallocSIMD.hpp:67]: please do not remove `XMallocSIMDConf::DefaultXMallocConfig`

File: src/include/mallocMC/reservePoolPolicies/CudaSetLimits.hpp [hunk 2]
```diff
@@ -45,7 +53,31 @@ namespace mallocMC
...
+            static auto classname() -> std::string
+            {
+                return "CudaSetLimits";
+            }
+        };
 
     } // namespace ReservePoolPolicies
 } // namespace mallocMC
+
+#endif
```
Comment (psychocoderHPC) [inline src/include/mallocMC/reservePoolPolicies/CudaSetLimits.hpp:83]: missing EOF

File: src/include/mallocMC/distributionPolicies/XMallocSIMD.hpp (hunk 1 above)
Comment (bernhardmgruber) [reply src/include/mallocMC/distributionPolicies/XMallocSIMD.hpp:67]: Well spotted, thx!

File: src/include/mallocMC/reservePoolPolicies/CudaSetLimits.hpp (hunk 2 above)
Comment (bernhardmgruber) [reply src/include/mallocMC/reservePoolPolicies/CudaSetLimits.hpp]: thx!

File: src/include/mallocMC/distributionPolicies/XMallocSIMD.hpp (hunk 1 above)
Comment (bernhardmgruber) [reply src/include/mallocMC/distributionPolicies/XMallocSIMD.hpp:67]: Well spotted, thx!

**Pull request merged**